examples/contract_template/README.md

```diff
@@ -0,0 +1,87 @@
+# Athena smart contract template
+
+This template shows the bare minimum that a smart contract must currently implement
+to be deployed and ran on the Athena-enabled test network.
+
+The contract implements 2 necessary methods:
+ - spawn
+ - verify
+
+Please check the comments in the [code](src/contract.rs) for explanations
+
+## Prerequisites
+### Rust installed
+The template is implememented in the Rust programming language and a working rust toolchain is needed. Follow the steps on https://rustup.rs too install it.
+
+### Athena toolchain
+It's needed to build the contract code to the Risc-V RV32EM target. 
+This target is not yet natively supproted by the Rust toolchain. 
+We maintain a toolchain at https://github.com/athenavm/rustc-rv32e-toolchain.
+
+This repository contains a `cargo athena` plugin used to build the contracts at [cli](../../cli) as well as `athup` helper script to simplify installing it.
+
+To install `cargo athena` download and execute the athup:
+```sh
+curl -L https://raw.githubusercontent.com/athenavm/athena/main/athup/athup | bash
+```
+
+or execute it directly from within this repository at `athup/athup`.
+
+
+## Building the contract
+To build, simply execute the below command. The built executable is placed at `elf/contract_template`.
+
+```sh
+cargo athena build
+```
+
+## Testing the contract
+The template comes with some tests in [tests](tests/test.rs). They can be run with
+
+```sh
+cargo test
+```
+
+Mind there is no `athena` in the command because the tests are built and executed directly on your machine. They spin up the Athena VM and execute the smart contract code inside.
+
+## Deploying the contract onto the Athena test network
+
+A contract can be deployed using the standard Spacemesh single-sig wallet. It has a `deploy` method designed specifically for this purpose. 
+First, you will need a singlesig wallet account with some funds in it. One SMH is more than enough.
```

Comment:
In one line you are using `single-sig` while in other one `singlesig` - please use only one from.